### ISSUE TITLE
Allow using multiple characters for each level.

### ISSUE THREAD

**Issue by dangom** [integral-dw/org-superstar-mode#23]
Does org-superstar-mode support using multiple characters for heading stars?
I'm trying to mimick Bear's behavior, as seen below:

<img width="491" alt="Screen Shot 2020-12-23 at 21 33 00" src="https://user-images.githubusercontent.com/7881842/106402331-0a8cbe80-63f7-11eb-9832-76cf73e5d2ef.png">

I tried changing the headline-bullets-alist but couldn't get it to do multiple characters. My skills with composition rules is also non-existent, and when I tried composing strings according to the rules of `compose-region`, as mentioned in the docstrings, all I manage to do is get two characters overlaid on top of each other.

Any advice would be welcome and thank you for sharing this lovely package!

**Comment by integral-dw** [comment]
Hi @dangom!  Interesting, I remember having played with the thought before, but never went through with looking too deeply into it. I'll have to play a little with this myself before being able to give a thorough answer, but I remember there was a technical issue why I didn't implement it immediately. I'll keep you updated!

**Comment by andersjohansson** [comment]
I was thinking in similar lines, wanting to have some more control over the display of bullets (specifically if it would be possible to add a face to the todo bullets). What I thought of then was to just use the `display` property instead of compose. Seemed to work (outdated and ugly [testing branch here](https://github.com/andersjohansson/org-superstar-mode/tree/display-testing), but I'm confused why it worked without adding `display` to `font-lock-extra-managed-props`). But I suppose there is a good reason for using compose over display that I have missed ([Maybe relating to indentation?](https://www.reddit.com/r/emacs/comments/keji66/what_is_bad_about_prettifysymbolsmode/gg36s81)).

**Comment by integral-dw** [comment]
Right, I forgot about the `display` property entirely, there was indeed something.  The fact you found this discussion regarding Prettify Symbols mode is no coincidence. A lot of the structural decisions made both in Org Superstar and Org Bullets strongly relates to Prettify Symbols, in fact that is why the internals use the prettify/unprettify nomenclature.   You can find a more elaborate explanation of the moving parts of Superstar [here](https://github.com/integral-dw/superstar-kit), where I outlined a recipe to create similar modes for different outline modes.  I hope to see things like `md-bullets` spring from it someday.
>Seemed to work (outdated and ugly testing branch here, but I'm confused why it worked without adding display to font-lock-extra-managed-props)

It also may be that `display` is already added to the properties Font Lock is bookkeeping by default, but that is mere conjecture.

**Comment by integral-dw** [comment]
I also find [this comment](https://www.reddit.com/r/emacs/comments/brt0sk/prettifysymbolsmode_is_so_cool/eomb5av?utm_source=share&utm_medium=web2x&context=3) particularly interesting, I believe this calls for another Issue until I had a talk with the Emacs mailing list.

**Comment by integral-dw** [comment]
Good news! It may be the case that this feature may become possible in the future, when I migrate from compose to display!
I will, however, have to put in a decent amount of legwork first.

**Comment by integral-dw** [comment]
Alright, the [volatile](https://github.com/integral-dw/org-superstar-mode/tree/volatile) branch has matured enough to soon be merged into the main branch.  What is missing is proper documentation for the assorted hooks. I will try and cook up a working example for you, though you will have to modify it to do exactly what you want (superstar has no way of counting the number of headlines as of now, although I am sure Org has a function for that)

**Comment by integral-dw** [comment]
Here's a working example: 
```lisp
(add-hook 'org-superstar-prettify-headline-hook
	  (lambda ()
	    (put-text-property (match-beginning 1) (match-end 1)
			       'display (format "#%d" (org-superstar-heading-level)))))
```
This would replace the leading bullet of a level *N* headline with "#N", so "#1", "#2", and so on. If you replace the heading level function with one that can compute the section numbering you want you can achieve any look you want.
